datastore/keycompatibility.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2011 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// This change adds additional compatibility to help customers
+// transition from google.golang.org/appengine/datastore to cloud.google.com/go/datastore.
+// Each lib google.golang.org/appengine/datastore and cloud.google.com/go/datastore contain the functions Key.Encode() and Key.Decode(). These functions
+// create base64 representations of a json marshalled type of datastore keys.  Customers have been using
+// these encoded values to communicate between services in appengine.  The protobuf key types
+// changed between google.golang.org/appengine/datastore and cloud.google.com/go/datastore making the corresponding base64 key strings incompatible.
+// Customers who attempt to upgrade to cloud.google.com/go/datastore that use this pattern will fail.
+// keycompatibility.go placed in google.golang.org/appengine/datastore enables forward compatibility of cloud.google.com/go/datastore encoded keys.
+// An update to cloud.google.com/go/datastore will also be necessary to enable backward compatibility.
+
+package datastore
+
+import (
+	"errors"
+	"sync"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/appengine/datastore/internal/keycompat"
+	"google.golang.org/appengine/internal"
+)
+
+var (
+	errKeyConversion = `Key conversions must be enabled in the application.
```

Comment:
errors.New(...)

also you probably don't want a multi-line string with all those tabs here

Reply:
With the move away from the struct, I dont think i need this error any longer.

Reply:
Decided to leave it.  Corrected per your suggestion.